project/frontend/src/services/firebase.js

```diff
@@ -35,6 +35,7 @@ export const onAuthStateChange = (callback) => {
         signedIn: true,
         name: user.displayName,
         email: user.email,
+        id: 1,
```

Comment:
We can't expect to receive an id here from firebase, as `id` is a concept/key that is in reference to our application database. Ideally we receive it in the response to our `POST` request to `"/authentication"` above.

Even more ideally - we shouldn't need our id to make calls to the backend after we are authenticated, as the backend should know who we are and what our id is. 

(IIRC there's also some bad practice stuff around exposing DB ids to clients, hence why [UUIDs](https://en.wikipedia.org/wiki/Universally_unique_identifier) are often used instead)

Reply:
Yeah I'm definitely aware of this. This is just the ID being hardcoded for now, when in reality it would be fetched from the backend. The reason it's hardcoded to 1 atm is because the last time I checked my own user (id=3) doesn't really have much data so there's nothing to display which is not very helpful for development. This will be fixed in issue #62

Reply:
It's worth discussion about why we might want to change it, but probably not worth actually doing currently.

My concern around returning the id here is building logic elsewhere that assumes it will be available at the time that callback is called, when that won't be the case in future. Really we're not 100% authenticated until the backend has confirmed our identity as well, so the callback should be after that happens.

Reply:
Haven't really thought very deeply about this yet, but at a glance I was thinking we fetch here? 
![Screen Shot 2021-02-08 at 9 20 50 pm](https://user-images.githubusercontent.com/62003343/107193342-9afc7c00-6a53-11eb-8fa5-d99efc8a83c1.png)
And use the call back in the .then()?

Reply:
Re-reading `services/firebase.js`, I think we should combine `signInWithGoogle` and `onAuthStateChange` into something like this:

```js
const signInWithGoogle = (callback) => {
  firebase.auth()
  .signInWithPopup(provider)
  .then(result => {
    let credential = result.credential;
    let token = credential.accessToken;

    let user = result.user;
    
    user.getIdToken(true)
      .then(tokenResult => {
        // POST idToken to /authentication, get id from response
        // call callback with user details and id
        // logged in!
      });
  }); 
}
```

WDYT?

Reply:
Refer to docs https://firebase.google.com/docs/auth/admin/manage-cookies#sign_in

Reply:
I tried this out quickly, and I seem to fall into an infinite loop of signing in and I'm not entirely sure why. Maybe @Phoebe-Khokgawe might understand this better, could you have a look at this?